data_adapter_oemof/calculations.py

```diff
@@ -34,3 +42,283 @@ def get_name(*args, counter=None):
+
+        Max values:
+        potential_annual_max
+        capacity_p_max
+        capacity_e_max
+        capacity_w_max
+
+        availability_timeseries_max
+        capacity_tra_connection_max
+        flow_share_max_<commodity>
+        sto_cycles_max
+        sto_max_timeseries
+
+        Returns
+        -------
+
+        """
+        max_value = 9999999999999
+        min_value = 0
+
+        min = ["capacity_p_min", "capacity_e_min", "capacity_w_min", "flow_share_min_"]
+
+        max = [
+            "potential_annual_max",
+            "capacity_p_max",
+            "capacity_e_max",
+            "capacity_w_max",
+            "availability_timeseries_max",
+            "capacity_tra_connection_max",
+            "flow_share_max_",
+            "sto_cycles_max",
+            "sto_max_timeseries",
+            "capacity_p_abs_new_max",
+            "capacity_e_abs_new_max",
+            "capacity_w_abs_new_max",
+        ]
+
+        for column in group_df.columns:
+            if column in ["method", "source", "comment", "bandwidth_type"]:
+                continue
+
+            """
+            Following is a check whether nans can be filled.
+
+            Commented check for columns that are faulty and need to be changed
+            Commented Error for incomplete columns as we dont know where it may cause errors yet
+
+            """
+            if column in max:
+                group_df[column] = group_df[column].fillna(max_value)
+            elif column in min:
+                group_df[column] = group_df[column].fillna(min_value)
+
+        return group_df
+
+    def find_and_replace_irrelevant_data(group_df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Finds and replaces irrelevant Data.
+
+        Searches for where investment is allowed
+            - If no Investment is allowed and there is no period above where investment was allowed
+                -> Data is replaced by pandas ffil/bfill
+            - If no Investment is allowed and there have been periods with investment before
+                -> Data is replaced by pandas ffil/bfill
+
+        Searches for decomissioned Processes
+            - If capacity of a process is 0 other nan data is replaced by mean.
+
+        Parameters
+        ----------
+        group_df
+
+        Returns
+        -------
+
+        """
+
+        capacity_columns = [
+            "capacity_p_inst_0",
+            "capacity_e_inst_0",
+            "capacity_w_inst_0",
+            "capacity_tra_inst_0",
+        ]
+
+        invest_zero = [
+            "capacity_p_abs_new_max",
+            "capacity_e_abs_new_max",
+            "capacity_w_abs_new_max",
+            "capacity_p_max",
+            "capacity_e_max",
+            "capacity_w_max",
+        ]
+        # no investment in decommissioning processes
+        # no capacity can be set on investment objects
+        if not any([x in capacity_columns for x in group_df.columns]):
```

Comment:
Kindly asking for opinion on if/else statement since it actually is not needed. 

I added it because data where capacity is set and investment is allowed at the same time the nans cannot be handled by the developed logic so far.